scripts/core/ArticlesListByQueryWithFilters.tsx

```diff
@@ -195,135 +201,212 @@ export class ArticlesListByQueryWithFilters extends React.PureComponent<IProps,
                             >
-                                <i className={`toggle-button__icon filetype-icon-${type}`} />
+                                {
+                                    icon != null
+                                        ? <i className={`toggle-button__icon ${icon}`} />
+                                        : label
+                                }
                             </button>
                         ))}
                     </div>
                 </div>
-                <SortBar
-                    sortOptions={getArticleSortOptions()}
-                    selected={this.state.sortOption}
-                    onSortOptionChange={(sortOption) => {
-                        this.setState({sortOption});
-                    }}
-                />
-            </div>
+            );
+        };
+
+        const sortFilterToolbar = (
+            <ResizeObserverComponent>
+                {(dimensions) => (
+                    <div
+                        style={{
+                            display: 'flex',
+                            height: toolbar2Height,
+                            alignItems: 'center',
+                            justifyContent: 'space-between',
+                            borderTop: '1px solid #d5d5d5',
+                            borderBottom: '1px solid #d5d5d5',
+                            paddingTop: 8,
+                            paddingBottom: 8,
+                            paddingLeft: padding,
+                            paddingRight: padding,
+                        }}
+                    >
+                        {getTypeFilteringComponent(dimensions.width < COMPACT_WIDTH)}
+
+                        <SortBar
+                            sortOptions={getArticleSortOptions()}
+                            selected={this.state.sortOption}
+                            onSortOptionChange={(sortOption) => {
+                                this.setState({sortOption});
+                            }}
+                        />
+                    </div>
+                )}
+            </ResizeObserverComponent>
         );
 
         return (
-            <MultiSelectHoc>
+            <MultiSelectHoc
+                shouldUnselect={(ids) => {
+                    // Use the same query except only for selected items.
+                    const queryForSpecificItems: ISuperdeskQuery = {
+                        ...query,
+                        page: 0,
+                        max_results: 200,
+                        filter: {
+                            $and: [
+                                query.filter,
+                                {'_id': {$in: ids.toJS()}},
+                            ],
+                        },
+                    };
+
+                    const elasticQuery = toElasticQuery(queryForSpecificItems);
+
+                    return httpRequestJsonLocal<IRestApiResponse<{_id: string}>>({ // projections only return _id
```

Comment:
don't we have some functions already to perform a search instead of using http manually?

Reply:
We do and this is one of those functions :smile:

There's also [dataApi](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/helpers/CrudManager.tsx#L135), but it uses mongo format for filtering. This one is more generic.

Reply:
I meant this is inside a render function inside a component, not inside a helper function to create requests

Reply:
This `httpRequestJsonLocal` call is <10 LOC. I don't think the helper that would be created out of this would be much simpler. For example, in this case only `_id` of items are returned which wouldn't be useful in other places.